utils/data-access.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * Middleware function for removing sensitive fields from the response object
+ * based on user roles and predefined rules.
+ *
+ * @param {Object} options - Options for configuring the middleware.
+ * @param {Array<Object>} options.rules - An array of rules specifying which fields
+ *   to remove based on allowed roles and key paths.
+ *   - `keyPath` {string}: A key path (e.g., 'sensitive.field1') representing the field
+ *     or subfield in the response object. You can use a wildcard (*) to match multiple
+ *     subfields (e.g., 'sensitive.*' to match all subfields under 'sensitive').
+ *   - `allowedRoles` {Array<string>}: An array of user roles that are allowed to access
+ *     the field specified by the `keyPath`. If the user has one of these roles, the field
+ *     will not be removed.
+ * @returns {Function} - Express middleware function.
+ *
+ * @example
+ * // Define rules for removing fields based on user roles
+ * const rules = [
+ *   {
+ *     allowedRoles: [SUPERUSER,MEMBER],
+ *     keyPath: 'sensitive.field1',
+ *   },
+ *   {
+ *     allowedRoles: [SUPERUSER],
+ *     keyPath: 'sensitive.field2',
+ *   },
+ *   {
+ *     allowedRoles: [SUPERUSER],
+ *     keyPath: 'sensitive.*', // Using a wildcard (*) to match all subfields under 'sensitive'
+ *   },
+ * ];
+ * // Use the middleware in your Express app
+ * app.use(dataAccessMiddleware({ rules }));
+ */
+const dataAccessMiddleware = (options = {}) => {
+  const rules = options.rules;
+
+  return (req, res, next) => {
+    try {
+      const { roles = {} } = req.userData;
+
+      const oldSend = res.send;
```

Comment:
Nit: What oldSend get?

can you help me to understand with this test case: https://github.com/Real-Dev-Squad/website-backend/pull/1564/files#diff-79f73954a237b17bfefc9e4db205ef4b93b4807163d96b0f2410cd599b1a5e27R34

Reply:
If you're asking about where the assertion is taking place then please check this [link](https://github.com/Real-Dev-Squad/website-backend/pull/1564/files#diff-79f73954a237b17bfefc9e4db205ef4b93b4807163d96b0f2410cd599b1a5e27R10-R16).

Reply:
> Nit: What oldSend get?

send method in res object which will later be used inside custom send method that we are creating.